lib/streamlit/web/server/component_request_handler.py

```diff
@@ -39,7 +39,14 @@ def get(self, path: str) -> None:
             return
 
         filename = "/".join(parts[1:])
-        abspath = os.path.join(component_root, filename)
+        # follow symlinks to get an accurate path
+        abspath = os.path.realpath(os.path.join(component_root, filename))
+
+        # Do NOT expose anything outside of the component root.
+        if os.path.commonprefix([component_root, abspath]) != component_root:
```

Comment:
Does this logic rely at all on `component_root` itself being an abspath when returns from `get_component_path`? (If `component_root` is a relative path, does anything break? And if so, should we add a test to ensure that `ComponentRegistry.get_component_path` returns an abspath?)

Reply:
The `component_root = os.path.realpath(component_root)` added above probably resolves this issue